homeassistant/components/demo/media_player.py

```diff
@@ -398,6 +407,18 @@ def set_repeat(self, repeat):
         self._repeat = repeat
         self.schedule_update_ha_state()
 
+    def join_players(self, group_members):
+        """Join `group_members` as a player group with the current player."""
+        self._group_members = [
+            self.entity_id,
+        ] + group_members
+        self.async_write_ha_state()
```

Comment:
In sync functions we need to use our sync api, `self.schedule_update_ha_state`. Or go back to use the coroutine function and async api.

Reply:
You are right, I have overlooked that. Strangely tests were passing... Anyway, it's fixed now.